content/kotsadm/updating/patching-with-kustomize.md

```diff
@@ -0,0 +1,93 @@
+---
+date: 2019-10-09
+linktitle: "Patching with Kustomize"
+title: Patching with Kustomize
+weight: 10040
+draft: false
+---
+
+[Kustomize](https://kustomize.io) allows for last-mile patches (i.e. modifications) to the app that are otherwise not configurable via `Config` page.
+
+## Directory Structure
+
+Click on `View files` and lets take a look at the directory structure.
+![Kustomize Dir Structure](/images/kustomize-dir-structure.png)
+
+### Upstream
+
+The `upstream` dir mirrors exactly the content pushed to a release. This includes the template functions, preflight checks, support-bundle, config options, license etc.
+
+**Note:** No changes should be made in this dir as they are overwritten on each new release.
+
+### Base
+
+After KOTS processes and renders the `upstream` it puts those files in the `base` dir. Any non-deployable manifests such as template functions, preflight checks, config options, etc. are removed and only the "deployable" application (i.e. `kubectl apply`) will be placed here.
+
+**Note:** No changes should be made in this dir as they are overwritten on each new release.
+
+### Overlays
+
+The `overlays` dir references the `base` dir and this is where your local kustomize patches should be placed. Any changes made here will persist between releases unlike `upstream` and `base`.
+
+* * *
+
+## Patching the Overlays
+
+Since `upstream` and `base` are ephemeral, lets make a kustomize patch in `overlays/downstreams/this-cluster/` so it persists between releases.
+
+Download the KOTS app locally.
+
+```shell
+kubectl kots download --namespace <namespace> --slug <app_slug> --dest ~/my-kots-app
```

Comment:
With these, rather than using `<namespace>` and `<app_slug>`, what do you think of making this more directly runnable by adding some env vars at the top? This way they don't have to copy and edit the long kotsadm command.

```shell
export NAMESPACE=some-ns
export APP_SLUG=some-slug
kubectl kots download --namespace ${NAMESPACE} ...
```

Alternatively, or in addition, what about educating the user how to get this command from kotsadm with a screenshot or two?